hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,547 @@
+		// TOCHECK as compared to: 
+		//descartes_core::RobotModelPtr model(new descartes_moveit::MoveitStateAdapter);
+		//TOCHECK or as in Godel
+		// init descartes with kinematic_state_ ? 
+		//   descartes_core::RobotModelPtr createRobotModel(const moveit::core::RobotStatePtr robot_state,
+                                                 // const std::string& group_name,
+                                                 // const std::string& tool_frame,
+                                                 // const std::string& world_frame,
+                                                 // uint8_t iterations)
+		if(descartes_robot_model_ptr_->initialize(ROBOT_DESCRIPTION_PARAM,
+		                              config_.group_name,
+		                              config_.world_frame,
+		                              config_.tip_link))
+		{
+		ROS_INFO_STREAM("Descartes Robot Model initialized");
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to initialize Robot Model");
+		exit(-1);
+		}
+
+		// TODO add planner type as config parameter?
+		bool succeeded_dense = dense_planner_.initialize(descartes_robot_model_ptr_);
+		if(succeeded_dense)
+		{
+		ROS_INFO_STREAM("Descartes dense planner initialized");
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to initialize Descartes dense Planner");
+		exit(-1);
+		}  
+
+		bool succeeded_sparse = sparse_planner_.initialize(descartes_robot_model_ptr_);
+		if(succeeded_sparse)
+		{
+		ROS_INFO_STREAM("Descartes sparse planner initialized");
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to initialize Descartes sparse Planner");
+		exit(-1);
+		}
+
+		ROS_INFO_STREAM("Descartes members initialized");
+
+		// creates publisher for trajectory visualization
+		marker_publisher_  = nh_.advertise<visualization_msgs::MarkerArray>(VISUALIZE_TRAJECTORY_TOPIC,1,true);
+
+		//Initializes MoveIt!
+		moveit_group_ptr_ = MoveGroupPtr(new moveit_group_interface::MoveGroup(config_.group_name));
+		//TOCHECK TODO doesn't compile
+		// home/ratnesh/projects/gsoc/src/hamilton/hamilton/src/hybrid_planner.cpp: In member function ‘void hamilton::HybridPlanner::init()’:
+		// /home/ratnesh/projects/gsoc/src/hamilton/hamilton/src/hybrid_planner.cpp:70:35: error: expected primary-expression before ‘(’ token
+		//    moveit_group_ptr_ = MoveGroupPtr(new moveit_group_interface::MoveGroup(config_.group_name));
+		//                                    ^
+		// /home/ratnesh/projects/gsoc/src/hamilton/hamilton/src/hybrid_planner.cpp:70:40: error: expected type-specifier before ‘moveit_group_interface’
+		//    moveit_group_ptr_ = MoveGroupPtr(new moveit_group_interface::MoveGroup(config_.group_name));
+		//                                         ^
+		// make[2]: *** [hamilton/hamilton/CMakeFiles/hamilton_demo.dir/src/hybrid_planner.cpp.o] Error 1
+		// make[1]: *** [hamilton/hamilton/CMakeFiles/hamilton_demo.dir/all] Error 2
+		// make: *** [all] Error 2
+		// Invoking "make -j8 -l8" failed
```

Comment:
You could use a IDE such as Eclipse or QtCreator.  There interfaces are rather user friendly and show you were the build error took place relatively well